src/api/spec/models/project_spec.rb

```diff
@@ -287,4 +287,118 @@
       }
     end
   end
+
+  describe '#deleted?' do
+    it 'returns false if the project exists in the app' do
+      expect(Project.deleted?(project.name)).to eq(false)
+    end
+
+    it 'returns false if _history file of project is empty' do
+      allow_any_instance_of(ProjectFile).to receive(:to_s).with(deleted: 1).and_return(nil)
+      expect(Project.deleted?('never-existed-before')).to eq(false)
+    end
+
+    it 'returns true if _history element has elements' do
+      allow_any_instance_of(ProjectFile).to receive(:to_s).with(deleted: 1).and_return(
+        "<revisionlist>\n  <revision rev=\"1\" vrev=\"\">\n    <srcmd5>d41d8cd98f00b204e9800998ecf8427e</srcmd5>\n    " \
+        "<version></version>\n    <time>1498113679</time>\n    <user>Admin</user>\n    <comment>1</comment>\n  " \
+        "</revision>\n</revisionlist>\n"
+      )
+
+      expect(Project.deleted?('never-existed-before')).to eq(true)
+    end
+  end
+
+  describe '#restore' do
+    let!(:admin_user) { create(:admin_user, login: 'Admin') }
+    let!(:project_name) { 'project_used_for_restoration' }
+    let(:project_meta) {
+      <<-PROJECT_META_XML
+        <project name=\"#{project_name}\">\n  <title></title>\n
+        <description></description>\n  <person userid=\"Admin\" role=\"maintainer\" />\n</project>
+      PROJECT_META_XML
+    }
+
+    before do
+      allow(Backend::Connection).to receive(:post).and_return(nil)
+      allow_any_instance_of(ProjectMetaFile).to receive(:to_s).and_return(project_meta)
+      allow(Collection).to receive(:find).and_return(ActiveXML::Node.new('<collection/>'))
+    end
+
+    describe 'verifys call to backend' do
+      it 'with option: user' do
+        expect(Backend::Connection).to receive(:post).with("/source/#{project_name}?cmd=undelete&user=#{admin_user}")
+        Project.restore(project_name, user: admin_user.login)
+      end
+
+      it 'with option: comment' do
+        expect(Backend::Connection).to receive(:post).with("/source/#{project_name}?cmd=undelete&comment=schaeuferle_mag_jeder")
+        Project.restore(project_name, comment: 'schaeuferle_mag_jeder')
+      end
+
+      it 'with options: user and comment' do
+        expect(Backend::Connection).to receive(:post).with("/source/#{project_name}?cmd=undelete&user=#{admin_user}&comment=schaeuferle_mag_jeder")
+        Project.restore(project_name, user: admin_user.login, comment: 'schaeuferle_mag_jeder')
+      end
+    end
+
+    describe 'project meta' do
```

Comment:
@mdeniz I agree with you. I should change it. In general I would also test the outcome of methods but as I said to you on Friday, I was not able to create a `deleted project` in the backend. (at least not in the test env) Maybe my decision was a bit driven by that... I'll gonna try it again because I also think it would be better to test this differently now.